app/src/main/java/org/xbmc/kore/utils/Task.java

```diff
@@ -0,0 +1,81 @@
+package org.xbmc.kore.utils;
+
+import android.support.annotation.NonNull;
+
+/**
+ * Represents a computation that will eventually complete.
```

Comment:
What's a computation? What does eventually complete mean? (Try to place yourself inside a Java only programmer)
I would just stick to the term Task as something that needs to be done.